packages/editor-ui/src/composables/useBeforeUnload.ts

```diff
@@ -0,0 +1,41 @@
+import { useCanvasStore } from '@/stores/canvas.store';
+import { useUIStore } from '@/stores/ui.store';
+import { useI18n } from '@/composables/useI18n';
+import { computed } from 'vue';
+import { VIEWS } from '@/constants';
+import type { useRoute } from 'vue-router';
+
+export function useBeforeUnload({ route }: { route: ReturnType<typeof useRoute> }) {
```

Comment:
Some documentation about what this is all about would be nice

Reply:
Good point! Added